### PR TITLE
Fix documentation for `write!` on `std::fmt` page

### LINKED ISSUE #36906: Incorrect documentation for `write!` in `std::fmt`
The [`std::fmt` docs](https://doc.rust-lang.org/std/fmt/index.html#write) currently say about the `write!` macro:

> Under the hood, this function is actually invoking the `write` function defined in this module.

But in fact, `write!` is defined as calling the `write_fmt` method on its argument, as per [its documentation](https://doc.rust-lang.org/std/macro.write!.html).

### PR DESCRIPTION
**Pull request by wesleywiser**
Fixes #36906

### CODE REVIEW

File: src/libcollections/fmt.rs [hunk 1]
```diff
@@ -261,7 +261,8 @@
 //! This and `writeln` are two macros which are used to emit the format string
 //! to a specified stream. This is used to prevent intermediate allocations of
 //! format strings and instead directly write the output. Under the hood, this
-//! function is actually invoking the `write` function defined in this module.
+//! function is actually invoking the `write_fmt` function defined on the
+//! `std::io::Write` trait.
```
Comment (GuillaumeGomez) [inline src/libcollections/fmt.rs]: While you're at it. Can you add urls for write_fmt and std::io::Write please?

Comment (wesleywiser) [inline src/libcollections/fmt.rs]: Sure! Can you check the urls? I'm not sure I added them correctly

**Comment by frewsxcv** [comment]
```
tidy check stage0 (x86_64-unknown-linux-gnu)
/build/src/libcollections/fmt.rs:264: trailing whitespace
```

**Comment by wesleywiser** [comment]
Fixed, thanks!

File: src/libcollections/fmt.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libcollections/fmt.rs]: Travis will tell us.

File: src/libcollections/fmt.rs [hunk 2]
```diff
@@ -261,7 +261,9 @@
 //! This and `writeln` are two macros which are used to emit the format string
 //! to a specified stream. This is used to prevent intermediate allocations of
 //! format strings and instead directly write the output. Under the hood, this
-//! function is actually invoking the `write` function defined in this module.
+//! function is actually invoking the
+//! [`write_fmt`](../../io/trait.Write.html#method.write_fmt) function defined on the
```
Comment (wesleywiser) [inline src/libcollections/fmt.rs]: It looks like the paths are wrong. I've tried `../io/trait.Write.html` and `../../io/trait.Write.html` but neither seem to work. @GuillaumeGomez any ideas?

Comment (GuillaumeGomez) [inline src/libcollections/fmt.rs]: The better thing to test is to build docs and then check by hand:

```
make docs
```

Then open `doc/index.html` with any web browser, go to the modified url and click on it. Then just compare with "real" url and you'll figure out easily.

Comment (wesleywiser) [inline src/libcollections/fmt.rs]: Thanks for the tip! When I try that, the links work correctly on my local computer but they still fail on Travis. Should I just remove them for now?

Comment (GuillaumeGomez) [inline src/libcollections/fmt.rs]: Yes, I'll take care of them. Once removed, please squash your commits.

Comment (wesleywiser) [inline src/libcollections/fmt.rs]: Done. Thank you!

**Pull request merged**